### PR TITLE
Add stop screens: dob mismatch, paper correction guidance

### PR DESCRIPTION
**Pull request by hepsimo**
- fix link on psc-dob-mismatch stop screen
- add general stop screen router and handler
- add env URLs with defaults for stop screen external links
- fix some screen content typos
- fix eslint config: switch case indent rule not matching code formatter indents

test cleanup:
- fix import CREATED_RESOURCE from pscVerification.mock
- rename individualStatementRouter.unit.ts as int test
- enable and fix skipped tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
         "coverage": "jest --coverage",
         "test": "jest",
         "test:coverage": "sh ./scripts/jest_coverage.sh",
-        "start": "npm run build && node dist/httpServer.js",
+        "start": "nodemon",
```
Comment (PhilipGlover) [inline package.json:20]: change to script added by mistake?

File: src/views/router_views/stop_screen/psc-dob-mismatch.njk
Comment (PhilipGlover) [inline src/views/router_views/stop_screen/psc-dob-mismatch.njk:1]: the govuk imports aren't needed here.  It's not part of your commit, but could be tidied here..

File: src/constants.ts [hunk 2]
```diff
@@ -13,7 +18,8 @@ export const Urls = {
     PERSONAL_CODE: `${urlWithTransactionIdAndSubmissionId}/individual/personal-code`,
     INDIVIDUAL_STATEMENT: `${urlWithTransactionIdAndSubmissionId}/individual/psc-statement`,
     PSC_VERIFIED: `${urlWithTransactionIdAndSubmissionId}/psc-verified`,
-    RLE_LIST: `${urlWithTransactionIdAndSubmissionId}/rle/rle-list`
+    RLE_LIST: `${urlWithTransactionIdAndSubmissionId}/rle/rle-list`,
+    STOP_SCREEN: `${urlWithTransactionIdAndSubmissionId}/stop/:stopType`
```
Comment (PhilipGlover) [inline src/constants.ts:22]: The Secure Company Stop screen I'm working on, comes before the Trans and Submission gets generated, so doesn't have this in the URL path.  Can we handle that??

File: src/views/router_views/stop_screen/psc-dob-mismatch.njk
Comment (hepsimo) [reply src/views/router_views/stop_screen/psc-dob-mismatch.njk:1]: Good spot, i'll remove the imports

File: src/config/index.ts [hunk 3]
```diff
@@ -140,5 +140,14 @@ export const env = readEnv(process.env, {
...
+        .default(""),
+    GET_RP01_LINK: url
+        .describe("Link to obtaining paper form RP01")
+        .default("https://www.gov.uk/government/publications/replace-a-document-not-meeting-requirements-rp01"),
+    GET_PSC01_LINK: url
+        .describe("Link to obtaining paper form PSC01")
+        .default("https://www.gov.uk/government/publications/give-notice-of-individual-person-with-significant-control-psc01"),
+    POST_TO_CH_LINK: url
+        .describe("Link to guidance on submission by post")
+        .default("https://www.gov.uk/government/news/posting-documents-to-companies-house")
```
Comment (PhilipGlover) [inline src/config/index.ts:152]: have you got PR's for these values in
- docker-chs-dev
- ecs-service-configs-dev

File: src/constants.ts (hunk 2 above)
Comment (hepsimo) [reply src/constants.ts:22]: Good spot, i'm making changes to allow for stop screens without those IDs in the URL

File: src/routers/handlers/stop-screen/stopScreenHandler.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import { PrefixedUrls, STOP_TYPE, toStopScreenPrefixedUrl } from "../../../constants";
+import { getLocaleInfo, getLocalesService, selectLang } from "../../../utils/localise";
+import { BaseViewData, GenericHandler, ViewModel } from "../generic";
+import { addSearchParams } from "../../../utils/queryParams";
+import { getUrlWithStopType, getUrlWithTransactionIdAndSubmissionId } from "../../../utils/url";
+import { env } from "../../../config";
+
+interface StopScreenHandlerViewData extends BaseViewData {
+    linkUrls?: string[];
+}
```
Comment (PhilipGlover) [inline src/routers/handlers/stop-screen/stopScreenHandler.ts:11]: For the Secure screen I need to pass the DSR  email address and telephone number.  I can pass them in this array, even though they are not link urls as such.
- so either I could add extra params to the ViewData
- or this array could be renamed to be more general??

File: src/constants.ts (hunk 2 above)
Comment (PhilipGlover) [reply src/constants.ts]: 👍

File: src/views/router_views/stop_screen/psc-dob-mismatch.njk
Comment (PhilipGlover) [reply src/views/router_views/stop_screen/psc-dob-mismatch.njk:1]: 👍

**Review by PhilipGlover — COMMENTED** [review]
- I have run this and accessed the 2 stops screens here.
- I have also merged my super-secure stop screen and accessed that with appropriate additions.
- Happy to approve, once:
  - conflicts resolved
  - question over name of `linkUrls` array decided
  - discussion had over the config items, whether it is enough to have them in the env map in `index.ts`

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/routers/handlers/stop-screen/stopScreenHandler.ts (hunk 4 above)
Comment (hepsimo) [reply src/routers/handlers/stop-screen/stopScreenHandler.ts:11]: I'll rename to `extraData`

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/routers/handlers/stop-screen/stopScreenHandler.ts (hunk 4 above)
Comment (PhilipGlover) [reply src/routers/handlers/stop-screen/stopScreenHandler.ts:11]: 👍

File: package.json (hunk 1 above)
Comment (hepsimo) [reply package.json:20]: Good spot i'll revert that.

Comment (PhilipGlover) [reply package.json]: 👍

**Review by PhilipGlover — APPROVED** [review]
as commented above, all issues now resolved and happy to approve.  Nice job.

**Pull request merged**